### PR TITLE
front: add e2e tests search and filter rolling stock

### LINKED ISSUE #7826: front : front: add e2e test filtering and searching for a rolling stock
Test details : https://github.com/osrd-project/osrd-tests/issues/47

### PR DESCRIPTION
**Pull request by Maymanaf**
The locked RS filter wasn't added in this test : Can't add a locked train for now to the list of the existing rolling stock
 ( To be determined later)

Closes : #7826

### CODE REVIEW

**Review by Uriel-Sautron — COMMENTED** [review]
Good work thank you !!!
I haven't tested it yet, i continue the review

File: front/tests/assets/scenario-utlis.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { type Page, expect } from '@playwright/test';
+import { v4 as uuidv4 } from 'uuid';
+
+import type { Infra } from 'common/api/osrdEditoastApi';
```
Comment (Uriel-Sautron) [inline front/tests/assets/scenario-utlis.ts]:
```suggestion
```

File: front/tests/assets/scenario-utlis.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import PlaywrightScenarioPage from '../pages/scenario-page-model';
+
+// Scenario creation
+export default async function createCompleteScenario(
+  page: Page,
+  trainScheduleName: string,
+  trainCount: string,
+  delta: string
+) {
+  const smallInfra = (await getInfra()) as Infra;
```
Comment (Uriel-Sautron) [inline front/tests/assets/scenario-utlis.ts]:
```suggestion
  const smallInfra = await getInfra();
```

**Review by Uriel-Sautron — COMMENTED** [review]
Good work, just a few comments

**Review by Uriel-Sautron — APPROVED** [review]
LGTM

File: front/tests/009-rollingstock-editor.spec.ts [hunk 3]
```diff
@@ -189,4 +200,92 @@ test.describe('Rollingstock editor page', () => {
...
+    // Verify that filtering reduces the count and all the RS have electic icons
+    expect(await rollingStockSelectorPage.getElectricRollingStockIcons.count()).toEqual(
+      await rollingStockSelectorPage.getRollingStockSearchNumber()
+    );
+
+    // Clear electric filter
+    await rollingStockSelectorPage.electricRollingStockFilter();
+
+    // Perform a filtering action for thermal rolling stock
+    await rollingStockSelectorPage.themalRollingStockFilter();
```
Comment (Caracol3) [inline front/tests/009-rollingstock-editor.spec.ts]: should be thermalRollingStockFilter

File: front/tests/009-rollingstock-editor.spec.ts [hunk 4]
```diff
@@ -189,4 +200,92 @@ test.describe('Rollingstock editor page', () => {
...
+    await rollingStockSelectorPage.electricRollingStockFilter();
+
+    // Verify that filtering reduces the count and all the RS have thermal and electric icons
+    expect(await rollingStockSelectorPage.getThermalElectricRollingStockIcons.count()).toEqual(
+      await rollingStockSelectorPage.getRollingStockSearchNumber()
+    );
+
+    // Clear filters
+    await rollingStockSelectorPage.electricRollingStockFilter();
+    await rollingStockSelectorPage.themalRollingStockFilter();
```
Comment (Caracol3) [inline front/tests/009-rollingstock-editor.spec.ts]: should be thermalRollingStockFilter

File: front/tests/pages/rolling-stock-selector-page.ts [hunk 5]
```diff
@@ -79,4 +103,19 @@ export default class RollingStockSelectorPage extends BasePage {
   async closeRollingstockModal() {
     await this.rollingStockSelectorModal.locator('.close').click();
   }
+
+  // Select Combustion engine RS filter
+  async themalRollingStockFilter() {
```
Comment (Caracol3) [inline front/tests/pages/rolling-stock-selector-page.ts]: should be thermalRollingStockFilter

**Review by Caracol3 — COMMENTED** [review]
just one letter missing in a fonction

**Review by Caracol3 — APPROVED** [review]
lgtm

File: front/tests/008-allowances.spec.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { test } from '@playwright/test';
 import { v4 as uuidv4 } from 'uuid';
 
-import createCompleteScenario, { allowancesManagement } from './assets/utils';
+import createCompleteScenario, { allowancesManagement } from './assets/scenario-utlis';
```
Comment (kmer2016) [inline front/tests/008-allowances.spec.ts]: Maybe we can create a directory to store different utility files organized by subject. For example :
```
assets
  utils
     scenario
     api
     ....
```
This implies migrating all functions from the utils.ts file into appropriately named files.
So We could create the new directory utils, then add scenario.ts file and move the utils.ts file inside the new one utils directory and rename it by index.ts (or another name) . And then we can plan to redispatch functions from the utils.ts file into folders with the right name in nother PR.

What do you think ?

File: front/tests/009-rollingstock-editor.spec.ts [hunk 7]
```diff
@@ -32,13 +41,15 @@ test.describe('Rollingstock editor page', () => {
     await findAndDeleteRollingStock(uniqueRollingStockName);
     await findAndDeleteRollingStock(uniqueUpdatedRollingStockName);
     await findAndDeleteRollingStock(uniqueDeletedRollingStockName);
+    await findAndDeleteRollingStock(thermalElectricRollingStockName);
```
Comment (kmer2016) [inline front/tests/009-rollingstock-editor.spec.ts]: Maybe extract these cleanup instructions into a function that can be called here and in `test.afterEach`.

File: front/tests/008-allowances.spec.ts (hunk 6 above)
Comment (Maymanaf) [reply front/tests/008-allowances.spec.ts]: After the drop V1 for STDCM the file scenario-utlis.ts will be removed and new clean test will be created
I just add it for easier clean up later so no need for a new directory

File: front/tests/009-rollingstock-editor.spec.ts (hunk 7 above)
Comment (Maymanaf) [reply front/tests/009-rollingstock-editor.spec.ts]: Done

File: front/tests/008-allowances.spec.ts (hunk 6 above)
Comment (Uriel-Sautron) [inline front/tests/008-allowances.spec.ts]:
```suggestion
import createCompleteScenario, { allowancesManagement } from './assets/scenario-utils';
```
Typo "utils"

Comment (kmer2016) [reply front/tests/008-allowances.spec.ts]: OK for `scenario-utils`. However, I believe we should create a utils directory to better organize our utility functions. This directory can be moved to the same level as assets and pages.

If that sounds good to you, could you please create an issue for this?

Thank you!

**Review by kmer2016 — APPROVED** [review]
LGTM !

**Pull request merged**